namelist.py

```diff
@@ -197,18 +204,34 @@ def dump(self, array_inline=True):
         return "\n".join(lines)
 
     def _format_value(self, value):
-        if isinstance(value, bool):
-            return value and '.true.' or '.false.'
-        elif isinstance(value, int):
-            return "%d" % value
-        elif isinstance(value, float):
-            return "%f" % value
-        elif isinstance(value, str):
-            return "'%s'" % value
-        elif isinstance(value, complex):
-            return "(%s,%s)" % (self._format_value(value.real), self._format_value(value.imag))
-        else:
-            raise Exception("Variable type not understood: %s" % type(value))
+        if sys.version_info < (3,0,0):  # if python2.x
```

Comment:
how about adding a variable that is a test for python3 and then using that in the `elif` below? Would reduce the copy-paste here

`is_python2 = sys.version_info < (3,0,0)`
...
`elif is_python2 and isinstance(value, unicode):`